### PR TITLE
Fix/2346 ajv error message prefix

### PR DESCRIPTION
**Pull request by chohmann**
Addresses #2346 

**Summary**

Passes along the already existing concept of validation context (input/output) to `convertAjvErrors` function to have the error message properly say it's in the Request or Response.

To test manually, I mocked a spec that returned either `bird` or `lizard` for pet type. Then called `prism proxy` with a spec that required type be either `cat` or `dog`  and configured it to forward requests to the prism mock server so that the response from the mock server would always be invalid for the spec used by the proxy.

**Checklist**

- The basics
  - [x] I tested these changes manually in my local or dev environment
- Tests
  - [x] Added or updated
  - [ ] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [ ] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [ ] N/A

**Screenshots**
Violation Logs
image

Violation Header
image

### CODE REVIEW

File: packages/http/src/validator/index.ts [hunk 1]
```diff
@@ -186,7 +200,7 @@ export const validateOutput: ValidatorFn<IHttpOperation, IHttpResponse> = ({ res
           )
         ),
         validateBody(element.body, response.contents || [], ValidationContext.Output, mediaType, bundle),
-        validateHeaders(element.headers || {}, response.headers || [], bundle)
+        validateHeaders(element.headers || {}, response.headers || [], ValidationContext.Output, bundle)
```
Comment (daniel-white) [inline packages/http/src/validator/index.ts:203]: was this an optional parameter? can we remove its optionality?

File: packages/http/src/validator/validators/utils.ts [hunk 2]
```diff
@@ -81,7 +87,7 @@ export const convertAjvErrors = (errors: NonEmptyArray<ErrorObject>, severity: D
       const path = prefix ? [prefix, ...errorPath] : errorPath;
       const errorPathType = errorPath.length > 0 ? (prefix == 'body' ? 'property ' : 'parameter ') : '';
       const errorSourceDescription =
-        'Request ' +
```
Comment (daniel-white) [inline packages/http/src/validator/validators/utils.ts:84]: is this whitespace important?

File: packages/http/src/validator/index.ts (hunk 1 above)
Comment (chohmann) [reply packages/http/src/validator/index.ts:203]: It wasn't a parameter to the header validation function before at all. I add it in this PR and it is not optional. It was only being passed to the body validator function before for some reason, and not the other validators 🤷‍♀️

Comment (daniel-white) [reply packages/http/src/validator/index.ts:203]: i see it now.

**Review by daniel-white — APPROVED** [review]

**Pull request merged**